FirebaseStorage/Tests/SwiftIntegration/StorageIntegration.swift

```diff
@@ -0,0 +1,678 @@
+      let setupExpectation = expectation(description: "setUp")
+
+      let largeFiles = ["ios/public/1mb"]
+      let emptyFiles =
+        ["ios/public/empty", "ios/public/list/a", "ios/public/list/b", "ios/public/list/prefix/c"]
+      setupExpectation.expectedFulfillmentCount = largeFiles.count + emptyFiles.count
+
+      do {
+        let bundle = Bundle(for: StorageIntegration.self)
+        let filePath = try XCTUnwrap(bundle.path(forResource: "1mb", ofType: "dat"),
+                                     "Failed to get filePath")
+        let data = try XCTUnwrap(try Data(contentsOf: URL(fileURLWithPath: filePath)),
+                                 "Failed to load file")
+
+        for largeFile in largeFiles {
+          guard let ref = storage?.reference().child(largeFile) else {
+            XCTFail("Failed to load: \(largeFile)")
+            continue
+          }
+          ref.putData(data, metadata: nil, completion: { _, error in
+            XCTAssertNil(error, "Error should be nil")
+            setupExpectation.fulfill()
+          })
+        }
+        for emptyFile in emptyFiles {
+          guard let ref = storage?.reference().child(emptyFile) else {
+            XCTFail("Failed to load: \(emptyFile)")
+            continue
+          }
+          ref.putData(Data(), metadata: nil, completion: { _, error in
+            XCTAssertNil(error, "Error should be nil")
+            setupExpectation.fulfill()
+          })
+        }
+
+      } catch {
+        XCTFail("Exception thrown setting up files in setUp")
+      }
+      waitForExpectations()
+    }
+  }
+
+  override func tearDown() {
+    app = nil
+    storage = nil
+    super.tearDown()
+  }
+
+  func testName() {
+    guard let app = app else {
+      XCTFail()
+      return
+    }
+    let aGS = app.options.projectID
+    let aGSURI = "gs://\(aGS!).appspot.com/path/to"
+    let ref = storage?.reference(forURL: aGSURI)
+    XCTAssertEqual(ref?.description, aGSURI)
+  }
+
+  func testUnauthenticatedGetMetadata() {
+    let expectation = self.expectation(description: "testUnauthenticatedGetMetadata")
+    let ref = storage?.reference().child("ios/public/1mb")
+    ref?.getMetadata(completion: { (metadata, error) -> Void in
+      XCTAssertNotNil(metadata, "Metadata should not be nil")
+      XCTAssertNil(error, "Error should be nil")
+      expectation.fulfill()
+    })
+    waitForExpectations()
+  }
+
+  func testUnauthenticatedUpdateMetadata() {
+    let expectation = self.expectation(description: "testUnauthenticatedUpdateMetadata")
+
+    let meta = StorageMetadata()
+    meta.contentType = "lol/custom"
+    meta.customMetadata = ["lol": "custom metadata is neat",
+                           "ちかてつ": "🚇",
+                           "shinkansen": "新幹線"]
+
+    let ref = storage?.reference(withPath: "ios/public/1mb")
+    ref?.updateMetadata(meta, completion: { metadata, error in
+      XCTAssertEqual(meta.contentType, metadata!.contentType)
```

Comment:
Should we also port the XCTAssertEqualObjects change to XCTAssertEqual back to the original test?

Reply:
Nope - Objective C NSStrings don't compare equal like Swift strings do